libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx

```diff
@@ -0,0 +1,135 @@
+
+const BoundariesSlider = ({
+     title,
+     marks,
+     boundariesData,
+     type,
+     max,
+   }: {
+  title: string;
+  marks: SliderMarks;
+  boundariesData: IBoundaries | undefined;
+  type: 'temperature' | 'humidity' | 'co2';
+  max: number;
+}) => {
+  const [dataTemp, setTempData] = useState<ITempData | null>();
+  const [dataHum, setHumData] = useState<IHumData | null>();
+  const [dataCO2, setCO2Data] = useState<ICO2Data | null>();
+  //const [data, setData] = useState<IBoundaries | null>();
+  const data = boundariesData;
+  const [api, contextHolder] = notification.useNotification();
+  const { trigger } = usePutRequest<unknown, IBoundaries>({
+    url: '/Terrarium/boundaries',
+    data: data,
+  });
+  const [message, setMessage] = useState<INotificationState>();
+  const handlePut = async () => {
+    console.log('Trigger');
+    try {
+      const response = await trigger();
+      console.log(response?.data);
+      return response;
+    } catch (error) {
+      console.error(error);
+      throw error;
+    }
+  };
+
+  const openNotification = (placement: NotificationPlacement, type: NotificationType) => {
+    api[type]({
+      message: message?.message,
+      placement,
+    });
+  };
+
+  const extractValues = (value: number | [number, number]): [number, number] => {
+    if (typeof value === 'number') {
+      return [value, value];
+    } else {
+      return value;
+    }
+  };
+
+  const onAfterChange = (value: number | [number, number]) => {
+    const [min, max] = extractValues(value);
+    if (data) {
+      if (type === 'temperature') {
+        setTempData({
+          tempMin: min,
+          tempMax: max,
+        });
+        data.temperatureBoundaryMin = min;
+        data.temperatureBoundaryMax = max;
+      } else if (type === 'humidity') {
+        setHumData({
+          humMin: min,
+          humMax: max,
+        });
+        data.humidityBoundaryMin = min;
+        data.humidityBoundaryMax = max;
+      } else {
+        setCO2Data({
+          co2Min: min,
+          co2Max: max,
+        });
+        data.cO2BoundaryMin = min;
+        data.cO2BoundaryMax = max;
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (dataTemp || dataHum || dataCO2) {
+      handlePut()
+          .then(response => {
+            if (response) {
+              setMessage({ message: 'Changes successfully saved!', type: 'success' });
+            } else {
+              setMessage({ message: 'Something went wrong!', type: 'error' });
+            }
+          })
+          .catch(error => {
+            console.error(error);
+            setMessage({ message: 'Something went very wrong!', type: 'error' });
+          });
+    }
+  }, [dataCO2, dataHum, dataTemp]);
+
+  useEffect(() => {
+    if (message) openNotification('bottomRight', message?.type);
+  }, [message]);
+
+  let defaultValue: [number, number] = [0, 50];
+
+  if (type === 'temperature') {
```

Comment:
Doesn't feel good if a code isn't in a hook or component, should be probably refactored